src/main/java/org/icgc/argo/workflow_management_scheduler/components/DirScheduler.java

```diff
@@ -110,28 +110,31 @@ private List<Run> getNextScheduledRuns(List<Run> activeRuns, List<Run> queuedRun
                   value -> {
                     val nextRunToInit = queuedRunsForWf.remove(queuedRunsForWf.size() - 1);
                     // update template params
-                    val templatedJson =
+                    val preTemplatedJson =
                         replaceTemplateWithValue(
                             nextRunToInit.getWorkflowParamsJsonStr(),
                             config.getWorkDirTemplate(),
-                            value);
+                            value.split(":")[0]);
+
+                    val templatedJson = addClusterParam(preTemplatedJson, value.split(":")[1]);
+
                     nextRunToInit.setWorkflowParamsJsonStr(templatedJson);
                     // set dirs
                     val newWorkDir =
                         replaceTemplateWithValue(
                             nextRunToInit.getWorkflowEngineParams().getWorkDir(),
                             config.getWorkDirTemplate(),
-                            value);
+                            value.split(":")[0]);
```

Comment:
We have `":"` hardcoded in quite a few places here. Can we introduce a global constant for `DIRECTORY_STRING_DELIMITER = ":"` or some other meaningful name that can be referenced throughout instead of hardcoding this string repeatedly?

It may even be better to have a function that splits a string from `nfs-external/nfs-1:cluster2` into an object with a directory and cluster property. Ideally we would put the logic for splitting a string into these properties in one place and not repeat the split logic throughout the code.